src/util.h

```diff
@@ -103,11 +103,8 @@ namespace rhost {
             using unique_ptr::operator=;
 
             protected_sexp& operator= (SEXP other) {
```

Comment:
This class should still have `protected_sexp& operator= (const protected_sexp& other)`, so please don't remove that. The `swap` change should take care of infinite recursion otherwise.

Also, it would be good to add ` operator= (protected_sexp&& other)` - it can just do `swap` as well. This will make containers of `protected_sexp` more efficient, because e.g. `vector` wouldn't add/remove references to every single element every time it has to resize.